src/main/java/org/spectra/cluster/model/consensus/GreedyConsensusSpectrum.java

```diff
@@ -341,4 +413,44 @@ public long getSummedPrecursorMz() {
     public int getSummedCharge() {
         return sumCharge;
     }
+
+    @Override
+    public int getPrecursorMz() {
+        return averagePrecursorMz;
+    }
+
+    @Override
+    public int getPrecursorCharge() {
+        return averageCharge;
+    }
+
+    @Override
+    public int[] getMzVector() {
+        return Arrays.stream(consensusPeaks).mapToInt(BinaryPeak::getMz).toArray();
+    }
+
+    @Override
+    public int[] getIntensityVector() {
+        return Arrays.stream(consensusPeaks).mapToInt(BinaryPeak::getIntensity).toArray();
+    }
+
+    @Override
+    public int getNumberOfPeaks() {
+        return consensusPeaks.length;
+    }
+
+    @Override
+    public String getUUI() {
+        return id;
+    }
+
+    @Override
+    public BinaryPeak[] getPeaks() {
+        return consensusPeaks;
+    }
+
+    @Override
+    public void setPeaks(BinaryPeak[] peaks) {
+        consensusPeaks = peaks;
```

Comment:
I guess that this function should not be supported. If you only set the consensusPeaks, next time the consensus spectrum is updated this change will be overwritten which basically leads to unexpected behavior. I suggest to throw an "UnsupportedOperationException" here.

Reply:
I have removed the setPeaks from the interface.